other/password_generator.py

```diff
@@ -2,34 +2,45 @@
 import string
 import random
 
-letters = [letter for letter in string.ascii_letters]
-digits = [digit for digit in string.digits]
-symbols = [symbol for symbol in string.punctuation]
-chars = letters + digits + symbols
-random.shuffle(chars)
 
-min_length = 8
-max_length = 16
-password = ''.join(random.choice(chars) for x in range(random.randint(min_length, max_length)))
-print('Password: ' + password)
-print('[ If you are thinking of using this passsword, You better save it. ]')
+def password_generator(max_length):
+    letters = [letter for letter in string.ascii_letters]
+    digits = [digit for digit in string.digits]
+    symbols = [symbol for symbol in string.punctuation]
+    chars = letters + digits + symbols
+    random.shuffle(chars)
+    password = ''.join(random.choice(chars)
+                       for x in range(max_length))
+    return password
```

Comment:
Really cool ideas here!  You sparked my imagination over my morning coffee.  Thanks.

I approve this PR in its current form _but_ what about...
```python
from random import choice
from string import ascii_letters, digits, punctuation

def password_generator(length=8):
    '''
    >>> len(password_generator())
    8
    >>> len(password_generator(length=16))
    16
    >>> len(password_generator(257))
    257
    >>> len(password_generator(length=0))
    0
    >>> len(password_generator(-1))
    0
    '''
    chars = tuple(ascii_letters) + tuple(digits) + tuple(punctuation)
    return ''.join(choice(chars) for x in range(length))
```
Some of the logic behind these suggestions:
* Change __max_length__ to just __length__ because it _completely_ determines the length of the output
* Give __length__ a sensible default value so that users can just fire-and-forget.  Perhaps this value should be higher
* Add doctests so that we can test the code by running __python3 -m doctest -v other/password_generator.py__
* tuples should be faster to create then lists and should take up slightly less memory
* We only need to randomize once.  __random.choice()__ should be a lighter weight operation than __random.shuffle()__
* __password_generator()__ should generate a password so there is no need to create a variable named __password__ only to get rid of it on the following line.